java/src/jmri/jmrit/operations/rollingstock/RollingStock.java

```diff
@@ -813,7 +815,7 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
                         if (e.getNewValue() != null) {
                             Location newLocation = locationManager.getLocationByReporter((jmri.Reporter) e.getNewValue());
                             if (newLocation != getLocation())
-                                setLocation(newLocation, null);
+                                setWhereLastSeen(newLocation);
```

Comment:
Paul,

 

Can you provide me with a copy of your script?  I’d like to pass it along to the user who requested these changes.

 

Dan

 

From: Paul Bender [mailto:notifications@github.com] 
Sent: Wednesday, August 10, 2016 4:14 PM
To: JMRI/JMRI <JMRI@noreply.github.com>
Cc: Daniel Boudreau <daboudreau@hotmail.com>; State change <state_change@noreply.github.com>
Subject: Re: [JMRI/JMRI] Add to rolling stock RFID "where last seen" and "when last seen" (#1767)

 

In java/src/jmri/jmrit/operations/rollingstock/RollingStock.java <https://github.com/JMRI/JMRI/pull/1767#discussion_r74320994> :

> @@ -813,7 +815,7 @@ public void propertyChange(java.beans.PropertyChangeEvent e) {
>                          if (e.getNewValue() != null) {
>                              Location newLocation = locationManager.getLocationByReporter((jmri.Reporter) e.getNewValue());
>                              if (newLocation != getLocation())
> -                                setLocation(newLocation, null);
> +                                setWhereLastSeen(newLocation);

Dan, 

1) Why are you duplicating the When Last Seen and Where Last Seen properties of the ID Tag? Those should be persistent values associated with the ID tag, so we can look those up, and shouldn't need to be stored with the rolling stock.

2) The code that sets Where Last Seen and When Last Seen instead of the Location and the Date also need to store the Location and the Date. We may not have ID tag readers at ever location, but when a car passes a reader, that reader is associated with a location, and we know the car MUST be at that location.

The 2nd part of this is VERY important to me, because it changes the previous behaviour. This code change breaks scripts that I've got locally to print wheel reports.

Paul

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub <https://github.com/JMRI/JMRI/pull/1767/files/b8a4457a349fa99fa532337fd1e13bf1b2ad22c2#r74320994> , or mute the thread <https://github.com/notifications/unsubscribe-auth/ANw2p8mEpqzBl6QcjCNzosfdeDl1OjOzks5qejEKgaJpZM4JhdU1> .  <https://github.com/notifications/beacon/ANw2p51kOJNkhZZT8hQwBJxQRCndu3NNks5qejEKgaJpZM4JhdU1.gif>